### PR TITLE
[now-dev] Update tests with verbose logging

### PR DESCRIPTION
**Pull request by AndyBitz**
Updates the tests for `now dev` to log more and make the hopefully more reliable.

### CODE REVIEW

File: packages/now-cli/test/dev/integration.js [hunk 1]
```diff
@@ -601,15 +697,14 @@ test('[now dev] 12-polymer-node', async t => {
...
-    const response = await result;
+    const response = await fetchWithRetry(`http://localhost:${port}`, 180);
 
     validateResponseHeaders(t, response);
 
     const body = await response.text();
     t.regex(body, /Polymer \+ Node.js API/gm);
   } finally {
-    dev.kill('SIGTERM');
+    await dev.kill('SIGTERM');
```
Comment (styfle) [inline packages/now-cli/test/dev/integration.js:707]: I noticed some tests use `testFixtureStdio()` and some do not. Why is that?

File: packages/now-cli/test/dev/integration.js [hunk 2]
```diff
@@ -916,45 +989,52 @@ test('[now dev] temporary directory listing', async t => {
...
 
 test('[now dev] add a `package.json` to trigger `@now/static-build`', async t => {
   const directory = fixture('trigger-static-build');
-  const { dev, port } = await testFixture(directory);
 
-  try {
-    dev.unref();
+  await fs.unlink(path.join(directory, 'package.json')).catch(() => null);
 
+  await fs
```
Comment (styfle) [inline packages/now-cli/test/dev/integration.js:1001]: Maybe `index.txt` can be a variable to be shared between this line and line 1018 in the script.

**Review by styfle — COMMENTED** [review]
```
Uncaught exception in test/dev/integration.js

  /home/circleci/repo/packages/now-cli/test/dev/integration.js:237

  Error: Failed for "06-gridsome".

 ╭ 
 │ Output for "06-gridsome"
 │ 
 │ ----- stdout -----
 │ Installing dependencies...
 │ 
 │ ----- stderr -----
 │ > Now CLI 16.7.1-canary.1 dev (beta) — https://zeit.co/feedback
 │ - Installing builders: @now/build-utils@canary
 │ > Creating initial build
 │ Gridsome v0.6.6
 │ 
 │ Initializing plugins...
 │ Initialize - 0.34s
 │ Load sources - 0s
 │ Create GraphQL schema - 0.06s
 │ Create pages and templates - 0.06s
 │ Generate code - 0.02s
 │ Bootstrap finish - 0.48s
 │ > Success! Build completed
 │ > Ready! Available at http://localhost:3022
 │ > Error! Failed to complete request to /: Error: connect ECONNREFUSED 127.0.0.1:38441
 ╰ 
  ✖ [now dev] 06-gridsome Rejected promise returned by test
  ✖ test/dev/integration.js exited with a non-zero exit code: 1

  Ran 1 test file out of 1 for job 1 of 1

  1 test failed
  1 uncaught exception

  [now dev] 06-gridsome

  /home/circleci/repo/packages/now-cli/node_modules/node-fetch/index.js:133

  Rejected promise returned by test. Reason:

  FetchError {
    code: 'ECONNRESET',
    errno: 'ECONNRESET',
    message: 'request to http://localhost:3022 failed, reason: socket hang up',
    name: 'FetchError',
    type: 'system',
  }
```
https://app.circleci.com/jobs/github/zeit/now/40547

File: packages/now-cli/test/dev/integration.js (hunk 1 above)
Comment (AndyBitz) [reply packages/now-cli/test/dev/integration.js:707]: `testFixtureStdio` was introduced later, and it doesn't provide a way to check the `stdio` of the `now dev` process, but some need to do that

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**